src/DIRAC/WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -246,10 +244,40 @@ def beginExecution(self):
                                             mode='Direct')
     if not result['OK']:
       return result
-    result = self.getQueues(result['Value'])
+    result = getQueuesResolved(siteDict=result['Value'],
+                               queueCECache=self.queueCECache,
+                               gridEnv=self.gridEnv,
+                               setup=gConfig.getValue('/DIRAC/Setup', 'unknown'),
+                               workingDir=self.workingDirectory,
+                               checkPlatform=self.checkPlatform,
+                               instantiateCEs=True)
     if not result['OK']:
       return result
 
+    self.queueDict = result['Value']
+    for queueName, queueDict in self.queueDict.items():
+
+      # Update self.sites
+      if queueDict['Site'] not in self.sites:
+        self.sites.append(queueDict['Site'])
+
+      # Update self.platforms, keeping entries unique and squashing lists
+      platform = queueDict['ParametersDict']['Platform']
```

Comment:
@aldbr This is failing in certification, I've not read it properly but can you take a look?

```
2021-08-19 08:10:26 UTC WorkloadManagement/SiteDirectorDteam ERROR: Agent exception while calling method beginExecution
Traceback (most recent call last):
  File "/opt/dirac/versions/v7.3.0a20-1629294914/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Core/Base/AgentModule.py", line 345, in am_secureCall
    result = functor(*args)
  File "/opt/dirac/versions/v7.3.0a20-1629294914/Linux-x86_64/lib/python3.9/site-packages/DIRAC/WorkloadManagementSystem/Agent/SiteDirector.py", line 266, in beginExecution
    platform = queueDict['ParametersDict']['Platform']
KeyError: 'Platform'
```

Reply:
Actually I think I know the solution: https://github.com/DIRACGrid/DIRAC/pull/5342